qbft/prepare.go

```diff
@@ -28,7 +28,12 @@ func (i *Instance) uponPrepare(signedPrepare *SignedMessage, prepareMsgContainer
 		return nil // no quorum yet
 	}
 
-	proposedRoot := i.State.ProposalAcceptedForCurrentRound.Message.Root
+	proposalMsgAccepted, err := DecodeMessage(i.State.ProposalAcceptedForCurrentRound.SSVMessage.Data)
```

Comment:
I've noticied that we are decoding the message multiple times around the code, maybe its worth to consider some approach where we decode once and pass the decoded message as well to functions to reduce the amount of calls to decode..